documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md

```diff
@@ -1,6 +1,136 @@
-# End-to-end encrypted messaging with Ockam and Kafka
+# End-to-end encrypted messaging with Kafka and Ockam Secure Channels
+
+## Background
+
+When using Apache Kafka and other stream processing services a common concern is data access and security.
+
+A lot of fields like finance and personal information handling require or advise data to be encrypted when stored and transferred.
+
+Since modern networks and data pipelines involve multiple steps when data is transferred or stored, it becomes more apparent that just encoding data in transit between two endpoints or while in storage does not provide enough security.
+
+This leads to emerging of End-to-End encryption as a preferred model for messaging.
+
+There were several attempts to define and implement end-to-end encryption for messaging system like Kafka, for example [KIP-317](https://cwiki.apache.org/confluence/display/KAFKA/KIP-317%3A+Add+end-to-end+data+encryption+functionality+to+Apache+Kafka)
+
+Many of those attemots are using some sort of a key storage and exchanging persistent encryption keys between devices. This means that the data is available to any party able to aquire the key.
```

Comment:
Most of these attempts are using some sort of key storage and exchange persistent keys between devices. A fundamental weakness of such a system is its vulnerability to any party able to acquire the key.